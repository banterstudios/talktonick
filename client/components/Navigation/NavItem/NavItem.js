import React, { Component } from 'react'
import glamorous from 'glamorous'
import FadeSlide from 'client/components/Transitions/FadeSlide'

const NavItemWrapper = glamorous.div(({ theme: { navPrimaryBg, navHighlightBg } }) => ({
  position: 'relative',
  padding: '10px 20px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center'
}))

class NavItem extends Component {
  render () {
    const { children } = this.props

    return (
      <NavItemWrapper>
        <FadeSlide>
          { children }
        </FadeSlide>
      </NavItemWrapper>
    )
  }
}

export default NavItem
