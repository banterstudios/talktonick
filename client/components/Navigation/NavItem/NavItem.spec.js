import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavItem from './NavItem'

describe('components/Navigation/NavItem', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<NavItem />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
