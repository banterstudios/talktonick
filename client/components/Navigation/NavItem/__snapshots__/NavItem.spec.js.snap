// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation/NavItem @renders should render correctly 1`] = `
<glamorous(div)>
  <FadeSlide />
</glamorous(div)>
`;
